src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -528,31 +503,31 @@ private ProtocolToken ProcessBlob(int frameSize)
         {
             int chunkSize = frameSize;
 
-            ReadOnlySpan<byte> availableData = _handshakeBuffer.ActiveReadOnlySpan;
-            // Discard() does not touch data, it just increases start index so next
-            // ActiveSpan will exclude the "discarded" data.
-            _handshakeBuffer.Discard(frameSize);
+            ReadOnlySpan<byte> availableData = _buffer.EncryptedReadOnlySpan;
+            // DiscardEncrypted() does not touch data, it just increases start index so next
+            // EncryptedSpan will exclude the "discarded" data.
+            _buffer.DiscardEncrypted(frameSize);
```

Comment:
I know this is exciting code. But I'm wondering now if there is possibility this could confuse some of the cleanup logic.